OpenRA.Mods.Common/ServerTraits/MasterServerPinger.cs

```diff
@@ -82,12 +82,22 @@ public void Tick(S server)
 					server.SendMessage(masterServerMessages.Dequeue());
 		}
 
-		public void ServerStarted(S server)
+		void INotifyServerStart.ServerStarted(S server)
 		{
 			if (server.Type != ServerType.Local && LanGameBeacon != null)
 				LanGameBeacon.Start();
 		}
 
+		void INotifyServerShutdown.ServerShutdown(S server)
+		{
+			var gameServer = new GameServer(server);
+			if (server.Settings.AdvertiseOnline)
+				UpdateMasterServer(server, gameServer.ToPOSTData(false));
+
+			if (LanGameBeacon != null)
```

Comment:
Shouldn't this be using `LanGameBeacon?.Stop();` the same as on `GameEnded`? It doesn't seem useful to advertise a closed game that nobody can join.

Reply:
Both approaches delist the game in a similar speed.